### PR TITLE
[BUG] Exclude pytest from running `datadownload` test by default

### LINKED ISSUE #7607: [BUG] Pytest Downloads datadownload marked tests
**Describe the bug**

I tried running the tests on my local sktime-dev environment and I ran `pytest`, it downloads a directory named `UnitTest/m5-forecasting-accuracy` containing a few `csv` files.

**Expected behavior**

`datadownload` test should be excluded by default when running `pytest`

**Versions**

pytest-8.3.4, pluggy-1.5.0 

System:
    python: 3.12.8 | packaged by Anaconda, Inc. | (main, Dec 11 2024, 16:31:09) [GCC 11.2.0]
executable: /home/jigyasu/anaconda3/envs/sktime-dev/bin/python3
   machine: Linux-6.8.0-50-generic-x86_64-with-glibc2.35

Python dependencies:
          pip: 24.2
       sktime: 0.35.0
      sklearn: 1.5.2
       skbase: 0.12.0
        numpy: 1.26.4
        scipy: 1.14.1
       pandas: 2.2.3
   matplotlib: 3.10.0
       joblib: 1.4.2
        numba: 0.60.0
  statsmodels: 0.14.4
     pmdarima: 2.0.4
statsforecast: 2.0.0
      tsfresh: None
      tslearn: None
        torch: None
   tensorflow: None

</details>

### PR DESCRIPTION
**Pull request by jgyasu**
#### Reference Issues/PRs

Fixes #7607 

#### What does this implement/fix? Explain your changes.

Exclude pytest from running `datadownload` test by default by adding a skip pytestmark variable

### CODE REVIEW

**Review by yarnabrina — CHANGES_REQUESTED** [review]
I posted a question in the issue regarding why is it being considered a bug, but more importantly I want to know how to run these tests if someone wants to. With a global pytestmark, I believe it'll be skipped every single time. So if someone wants to unskip, we should document what do they have to do?

Also, there are already Makefile target available to run all by skipping these tests, so I want to understand when is this change useful.
